packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart

```diff
@@ -0,0 +1,381 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:flutter/foundation.dart';
+
+import 'actions.dart';
+import 'framework.dart';
+import 'shortcuts.dart';
+import 'text_editing_intents.dart';
+
+/// A [Shortcuts] widget with the shortcuts used for the default text editing
+/// behavior.
+///
+/// This default behavior can be overridden by placing a [Shortcuts] widget
+/// lower in the widget tree than this. See [DefaultTextEditingActions] for an example
+/// of remapping a text editing [Intent] to a custom [Action].
+///
+/// {@tool snippet}
+///
+/// This example shows how to use an additional [Shortcuts] widget to override
+/// some default text editing keyboard shortcuts to have new behavior. Instead
+/// of moving the cursor, alt + up/down will instead change the focused widget.
+///
+/// ```dart
+/// @override
+/// Widget build(BuildContext context) {
+///   return Scaffold(
+///     body: Center(
+///       child: Shortcuts(
+///         shortcuts: <LogicalKeySet, Intent>{
+///           LogicalKeySet(LogicalKeyboardKey.alt, LogicalKeyboardKey.arrowDown): NextFocusIntent(),
+///           LogicalKeySet(LogicalKeyboardKey.alt, LogicalKeyboardKey.arrowUp): PreviousFocusIntent(),
+///         },
+///         child: Column(
+///           children: <Widget>[
+///             TextField(
+///               decoration: InputDecoration(
+///                 hintText: 'alt + down moves to the next field.',
+///               ),
+///             ),
+///             TextField(
+///               decoration: InputDecoration(
+///                 hintText: 'And alt + up moves to the previous.',
+///               ),
+///             ),
+///           ],
+///         ),
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// {@tool snippet}
```

Comment:
@gspencergoog Examples like this won't work on web.  Does Shortcuts just not work at all on web?  Is there anything I can do to reduce this divergence in this PR?

Reply:
Since you force-pushed, I can't see what example you're referring to (GitHub just says it "looked everywhere" for the changes). Shortcuts definitely should be working on the web.  For instance, [this example](https://master-api.flutter.dev/flutter/widgets/Shortcuts-class.html#widgets.Shortcuts.1) works in Dartpad.

Reply:
@justinmc @gspencergoog I think this is in reference to Shortcuts in the context of Text Editing. Can confirm that shortcuts work fine throughout our web app. We're using them quite a bit.

However, I'm not sure if it would be possible to override TextEditing shortcuts on web?

Reply:
Playing with that example, it seems like if I add a TextField to the page, it will swallow all keyboard Shortcuts when it has focus.  This appears to be specific to web, because if I try invoking a keyboard shortcut on Mac running master with a TextField focused, it works.

This is going to be a problem for the reasons mentioned by @venkatd.  Also, @shihaohong will likely run into problems with this when trying to create keyboard shortcuts for Autocomplete.  I'll try to get an opinion from someone working on web.

Reply:
@justinmc in our Flutter web app *only* keyboard Shortcuts that are used by the TextField are swallowed. For example `LogicalKeyboardKey.arrowDown` and `LogicalKeyboardKey.up` are swallowed while `LogicalKeySet(LogicalKeyboardKey.control, LogicalKeyboardKey.keyJ)` are properly handled even when a TextField is focused.

Meanwhile I can confirm that when running a MacOS build, everything works perfectly.

Reply:
@mdebbar Do you know why/how web is swallowing Shortcuts when a text field is focused?

I'm worried that this PR is going to make web into a special case for handling keyboard events and wanted to run this by you to see what I can do.

Reply:
Here's a Dartpad showing what I mean: https://dartpad.dartlang.org/b17b1acce8bb9d3f9951981a93f80084?null_safety=true

On web, the Shortcut is ignored.  Nothing is logged, and the cursor moves like normal.  If you copy this code and run it locally, you'll see that when the left arrow is pressed, the cursor doesn't move, and the print statement is logged.

Reply:
Discussed this offline and our solution is here: 00c03ad

Web will explicitly ignore all of these text editing shortcuts and allow them to be handled by the web engine.  However, it's possible to override these by using a Shortcuts widget placed anywhere in the widget tree.

@shihaohong This is good news for keyboard shortcuts for Autocomplete on web.  If you just use a Shortcuts widget like normal, it should work identically on the web.